File: lab1/SimUDuck/DecoyDuck.cpp
#include "stdafx.h"
#include "DecoyDuck.h"

DecoyDuck::DecoyDuck()
	: Duck(std::make_unique<FlyNoWay>(), std::make_unique<MuteQuackBehavior>())
{
}

void DecoyDuck::Display() const
{
	std::cout << "I'm decoy duck" << std::endl;
}

File: lw1/src/duck/DecoyDuck.php
<?php

class DecoyDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(new FlyNoWay(), new MuteQuackBehavior(), new NoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m decoy duck' . PHP_EOL;
    }

    public function dance(): void
    {
    }
}